halogen/types.py

```diff
@@ -280,16 +310,56 @@ def deserialize(self, value, **kwargs):
 class Nullable(Type):
     """Nullable type."""
 
-    def __init__(self, nested_type, *args, **kwargs):
+    def __init__(self, nested_type: Type, *args, **kwargs):
         self.nested_type = nested_type
         super(Nullable, self).__init__(*args, **kwargs)
 
-    def serialize(self, value, **kwargs):
+    def serialize(self, value: Optional[Any], **kwargs):
         if value is None:
             return None
         return self.nested_type.serialize(value, **kwargs)
 
-    def deserialize(self, value, **kwargs):
+    def deserialize(self, value: Optional[Any], **kwargs):
         if value is None:
             return None
         return self.nested_type.deserialize(value, **kwargs)
+
+
+class Enum(Type):
+    """Enum schema type for enum.Enum."""
+
+    def __init__(
+        self,
+        enum_type: type[enum.Enum],
+        use_values: bool = False,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        if not issubclass(enum_type, enum.Enum):
+            raise TypeError("PythonEnum schema only support enum.Enum.")
+        self.enum_type = enum_type
+        self.use_values = use_values
+
+    def serialize(self, value: Optional[enum.Enum], **kwargs):
+        if value is None:
+            raise ValueError("None passed, use Nullable type for nullable values")
+
+        if self.use_values:
+            value = value.value
+        else:
+            value = value.name
+        return super().serialize(value, **kwargs)
+
+    def deserialize(self, value: Optional[str], **kwargs):
+        if value is None:
+            raise ValueError("None passed, use Nullable type for nullable values")
+
+        try:
+            if self.use_values:
+                value = next(item for item in self.enum_type if item.value == value)
+            else:
+                value = next(item for item in self.enum_type if item.name == value)
+            return super().deserialize(value, **kwargs)
+        except StopIteration:
```

Comment:
with my suggestion, a StopIteration will never be raised anymore, so you can remove this (it raises a ValueError instead).

Reply:
✅